src/socket.rs

```diff
@@ -368,6 +368,22 @@ impl Socket {
         self.inner.set_ttl(ttl)
     }
 
+    /// Set the value of IP_BOUND_IF option on this socket.
+    ///
+    /// See libc::if_nametoindex on mapping interface name to interface index.
+    #[cfg(all(feature = "all", not(target_os = "redox")))]
```

Comment:
This should be `#[cfg(all(feature = "all", target_vendor = "apple"))]` as its only available on Apple platforms from what I can gather from the pr.